### PR TITLE
Allow Testing Of Withdrawals On E2E Mainnet

### PR DESCRIPTION
**Pull request by nisdas**
**What type of PR is this?**

Bug Fix

**What does this PR do? Why is it needed?**

This fixes our e2e setup to allow withdrawals to be tested on mainnet. We allow the configuring of the withdrawal sweep if we are running a single-client e2e run.

**Which issues(s) does this PR fix?**

N.A

**Other notes for review**

### CODE REVIEW

**Review by saolyn — DISMISSED** [review]

**Review by saolyn — DISMISSED** [review]

File: testing/endtoend/evaluators/operations.go [hunk 1]
```diff
@@ -569,7 +569,7 @@ func submitWithdrawal(ec *e2etypes.EvaluationContext, conns ...*grpc.ClientConn)
 	changes := make([]*v2.SignedBLSToExecutionChange, 0)
 	// Only send half the number of changes each time, to allow us to test
 	// at the fork boundary.
-	wantedChanges := numOfExits / 2
```
Comment (rkapka) [inline testing/endtoend/evaluators/operations.go:572]: Why change this?

File: testing/endtoend/evaluators/operations.go [hunk 2]
```diff
@@ -92,8 +93,14 @@ var SubmitWithdrawal = e2etypes.Evaluator{
 
 // ValidatorsHaveWithdrawn checks the beacon state for the withdrawn validator and ensures it has been withdrawn.
 var ValidatorsHaveWithdrawn = e2etypes.Evaluator{
-	Name:       "validator_has_withdrawn_%d",
-	Policy:     policies.OnEpoch(helpers.CapellaE2EForkEpoch + 1),
+	Name: "validator_has_withdrawn_%d",
+	Policy: func(currentEpoch primitives.Epoch) bool {
+		// Use a more complex way to determine the policy due to differences
+		// in seed lookahead in mainnet and minimal config.
```
Comment (prestonvanloon) [inline testing/endtoend/evaluators/operations.go:99]: Is this a TODO? I don't understand this comment here.

File: testing/endtoend/evaluators/operations.go [hunk 3]
```diff
@@ -613,9 +620,6 @@ func submitWithdrawal(ec *e2etypes.EvaluationContext, conns ...*grpc.ClientConn)
 func validatorsAreWithdrawn(ec *e2etypes.EvaluationContext, conns ...*grpc.ClientConn) error {
 	// We skip this for multiclient runs as lighthouse does not have the ability
 	// to configure the withdrawal sweep for the end to end test.
```
Comment (prestonvanloon) [inline testing/endtoend/evaluators/operations.go:622]:
```suggestion
```
Seems like you should delete this comment too?

File: testing/endtoend/mainnet_e2e_test.go [hunk 4]
```diff
@@ -10,10 +10,10 @@ import (
 
 // Run mainnet e2e config with the current release validator against latest beacon node.
 func TestEndToEnd_MainnetConfig_ValidatorAtCurrentRelease(t *testing.T) {
-	r := e2eMainnet(t, true, false, types.StartAt(version.Phase0, params.E2EMainnetTestConfig()))
+	r := e2eMainnet(t, true, false, types.StartAt(version.Phase0, params.E2EMainnetTestConfig(), false))
```
Comment (prestonvanloon) [inline testing/endtoend/mainnet_e2e_test.go]: I think adding this inline comment helps the reader understand this parameter. Ideally, all of the "true" and "false" values in this method call would have the inline so the reader doesn't have to go look it up. 

```suggestion
	r := e2eMainnet(t, true, false, types.StartAt(version.Phase0, params.E2EMainnetTestConfig(), false /*multiclient*/))
```

File: testing/endtoend/mainnet_e2e_test.go [hunk 5]
```diff
@@ -10,10 +10,10 @@ import (
...
 	r.run()
 }
 
 func TestEndToEnd_MainnetConfig_MultiClient(t *testing.T) {
-	e2eMainnet(t, false, true, types.StartAt(version.Phase0, params.E2EMainnetTestConfig()), types.WithValidatorCrossClient()).run()
+	e2eMainnet(t, false, true, types.StartAt(version.Phase0, params.E2EMainnetTestConfig(), true), types.WithValidatorCrossClient()).run()
```
Comment (prestonvanloon) [inline testing/endtoend/mainnet_e2e_test.go]:
```suggestion
	e2eMainnet(t, false, true, types.StartAt(version.Phase0, params.E2EMainnetTestConfig(), true /*multiclient*/), types.WithValidatorCrossClient()).run()
```

File: testing/endtoend/mainnet_scenario_e2e_test.go [hunk 6]
```diff
@@ -9,7 +9,7 @@ import (
 )
 
 func TestEndToEnd_MultiScenarioRun_Multiclient(t *testing.T) {
-	runner := e2eMainnet(t, false, true, types.StartAt(version.Phase0, params.E2EMainnetTestConfig()), types.WithEpochs(22))
+	runner := e2eMainnet(t, false, true, types.StartAt(version.Phase0, params.E2EMainnetTestConfig(), true), types.WithEpochs(22))
```
Comment (prestonvanloon) [inline testing/endtoend/mainnet_scenario_e2e_test.go]:
```suggestion
	runner := e2eMainnet(t, false, true, types.StartAt(version.Phase0, params.E2EMainnetTestConfig(), true /*multiclient*/), types.WithEpochs(22))
```

**Review by terencechain — APPROVED** [review]

**Review by rkapka — APPROVED** [review]

**Pull request merged**